pkg/resource/stack/secrets.go

```diff
@@ -146,28 +178,81 @@ func (csm *cachingSecretsManager) BulkDecrypt(ctx context.Context, ciphertexts [
 	return csm.decrypter.Value().BulkDecrypt(ctx, ciphertexts)
 }
 
-// encryptSecret encrypts the plaintext associated with the given secret value.
-func (csm *cachingSecretsManager) encryptSecret(ctx context.Context,
-	secret *resource.Secret, plaintext string,
-) (string, error) {
+func (csm *cachingSecretsManager) EnqueueEncryption(ctx context.Context,
```

Comment:
Is there a possibility of this being called in parallel?  Do we need any locking around the `encryptQueue` for that?